lm_eval/evaluator.py

```diff
@@ -84,9 +88,11 @@ def simple_evaluate(model, model_args=None, tasks=[],
 
     return results
 
+decontaminate_suffix = "_decontaminate"
 
 @positional_deprecated
-def evaluate(lm, task_dict, provide_description=None, num_fewshot=0, limit=None, bootstrap_iters=100000, description_dict=None):
+def evaluate(lm, task_dict, provide_description=None, num_fewshot=0, limit=None, bootstrap_iters=100000, description_dict=None,
```

Comment:
I think the things in `evaluate` would be a lot clearer if the decontamination code were extracted into a separate class with some hooks

Reply:
the class would just do no-ops when decontaminate is turned off so we dont need `if decontanimate` everywhere in `evaluate`

Reply:
I agree that evaluate is not particularly pleasant currently, but still prefer if statements over global disabling and no-ops. If this is a big stylistic issue please refactor and we will review.